File: TOP10majorSorting/selectionsort.py
def selection_sort(nums):
    n=len(nums)
    for i in range(n-1):
        min_id=i
        for j in range(i+1,n):
            if nums[j]<nums[min_id]:
                min_id=j
        nums[i],nums[min_id]=nums[min_id],nums[i]
    return nums
nums=[8,-4,2,-1,0]
ans=selection_sort(nums.copy())
print (ans)

File: TOP10majorSorting/selectionsort.cpp
#include <iostream>
#include <vector>
using namespace std;
void selectionsort(vector<int> &nums)
{
    int n = nums.size();
    for (int i = 0; i < n - 1; i++)
    {
        int min_id = i;
        for (int j = i + 1; j < n; j++)
        {
            if (nums[j] < nums[min_id])
            {
                min_id = j;
            }
        }
        swap(nums[i], nums[min_id]);
    }
}
int main(void)
{
    vector<int> nums = {8, -4, 2, -1, 0};
    selectionsort(nums);
    for (int num : nums)
    {
        cout << num << " ";
    }
    cout << endl;
    return 0;
}